src/convert_handle.cpp

```diff
@@ -0,0 +1,59 @@
+#include "convert_handle.hpp"
+namespace vg {
+using namespace std;
+    void convert_handle_graph(const HandleGraph* converting, MutableHandleGraph* converted) {
+        assert(converted->node_size() == 0);
+        
+        // If xg is a null pointer, throw a runtime error
+        if (converting == nullptr) {
+            throw runtime_error("There is no graph to convert");
+        }
+        // Iterate through each handle in xg and create the same handle in mutable graph
+        converting->for_each_handle([&](const handle_t& here) {
+            // Get the id of the graph handle
+            id_t converting_id = converting->get_id(here);
+            // Get the sequence of the graph handle
+            string converting_seq = converting->get_sequence(here);
+            // Create a handle in mutable graph using the graph id and sequence
+            converted->create_handle(converting_seq, converting_id);
+        });
+        // add any edges that are not yet present
+        converting->for_each_edge([&](const edge_t& edge) {
+            edge_t converted_edge(converted->get_handle(converting->get_id(edge.first), converting->get_is_reverse(edge.first)),
+                             converted->get_handle(converting->get_id(edge.second), converting->get_is_reverse(edge.second)));
+            if (!converted->has_edge(converted_edge)) {
```

Comment:
I think this check should be unnecessary. The implementation of `for_each_edge` guarantees that each edge is iterated over only once, so if you have a well-formed graph going in, this should never evaluate to false